src/osd_field.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "config/rcxml.h"
+#include "view.h"
+#include "workspaces.h"
+#include "labwc.h"
+#include "osd.h"
+
+/* includes '%', terminating 's' and NULL byte, 8 is enough for %-9999W */
+#define LAB_FIELD_SINGLE_FMT_MAX_LEN 8
+static_assert(LAB_FIELD_SINGLE_FMT_MAX_LEN <= 255, "fmt_len is a unsigned char");
+
+/* forward declares */
+typedef void field_set_fn_t(struct buf *buf, struct view *view, const char *format);
```

Comment:
We haven't used `typedef` for functions before.

https://www.kernel.org/doc/html/v4.10/process/coding-style.html#typedefs
https://git.sr.ht/~sircmpwn/cstyle#features-to-avoid

Might a simple `void (fn *)(struct buf *buf, struct view *view, const char *format)` suffice in the struct below.

Reply:
https://git.sr.ht/~sircmpwn/cstyle#variable-declarations states that typedefs `MUST` be used for function pointers that are arguments for other functions and also states later
> Typedefs are usually a bad idea, but function pointers are so verbose and weird and difficult to read that typedefs help out here.

Reply:
You’re right. I stand corrected. Happy with approach.